src/main/java/org/jenkinsci/plugins/scm_filter/GitHubPullRequestSkipTrait.java

```diff
@@ -94,18 +94,14 @@ public boolean isExcluded (
         ) throws IOException, InterruptedException {
 
             if (scmHead instanceof PullRequestSCMHead) {
-                Iterable<GHPullRequest> pulls = ((GitHubSCMSourceRequest) scmSourceRequest).getPullRequests();
-                Iterator<GHPullRequest> pullIterator = pulls.iterator();
-                String prName = scmHead.getName();
-
-                while (pullIterator.hasNext()) {
-                    GHPullRequest pull = pullIterator.next();
-
-                    if (prName.equals("PR-" + pull.getNumber())) {
-                        String title = pull.getTitle().toLowerCase();
-                        return title.contains("[ci skip]") || title.contains("[skip ci]") || title.contains("[wip]");
-                    }
-                }
+                GHRepository repository = ((GitHubSCMSourceRequest) scmSourceRequest).getRepository();
+                int prNumber = Integer.parseInt(scmHead.getName().substring(3));
```

Comment:
Add comment on why we do substring and give example.

Reply:
Names of pull requests are formatted as "PR-<number>".

Will include an inline comment to clarify.

Reply:
Done!

Reply:
Done!